tests/Clinic.UnitTests/ServiceTests/OfficeServiceTests.cs

```diff
@@ -1,13 +1,183 @@
+            _officeRepository.GetByIdAsync(id, default).ReturnsNull();
+
+            //Act
+            var result = await _officeService.GetByIdAsync(id, default);
+
+            //Assert
+            await _officeRepository.Received(1).GetByIdAsync(id, default);
+            result.ShouldBeNull();
+        }
+
+        [Fact]
+        public async Task GetAllAsync_ByIsActive_ReturnCollectionOfOffices()
+        {
+            //Arrange
+            var office = _mapper.Map<IEnumerable<OfficeEntity>>(TestOfficeModels.SortOffices(null, null, OfficeStatus.Active, default));
+
+            _officeRepository.GetAllAsync(null, null, OfficeStatus.Active, Arg.Any<CancellationToken>()).Returns(office);
+
+            //Act
+            var result = await _officeService.GetAllAsync(null, null, OfficeStatus.Active, new CancellationToken());
+
+            //Assert
+            await _officeRepository.Received(1).GetAllAsync(null, null, OfficeStatus.Active, Arg.Any<CancellationToken>());
+            result.Count().ShouldBeEquivalentTo(1);
+        }
+
+
+        [Fact]
+        public async Task CreateAsync_ReturnCreatedOffice()
+        {
+            //Arrange
+            var officeEntity = TestOfficeModels.Office;
+            var officeModel = _mapper.Map<Office>(officeEntity);
+
+            _officeRepository.AddAsync(Arg.Any<OfficeEntity>(), default).Returns(officeEntity);
+
+            //Act
+            var result = await _officeService.CreateAsync(officeModel, default);
+
+            //Assert
+            result.Address.ShouldBeEquivalentTo(officeEntity.Address);
+        }
+
+        [Fact]
+        public async Task CreateModel_InvalidModel_ShouldNotResiveRequest()
+        {
+            //Arrange
+            var officeEntity = TestOfficeModels.Office;
+            var officeModel = _mapper.Map<Office>(officeEntity);
+
+            _officeRepository.AddAsync(Arg.Any<OfficeEntity>(), default).ReturnsNull();
+
+            //Act
+            var action = async () => await _officeService.CreateAsync(officeModel, default);
+
+            //Assert
+            await _officeRepository.DidNotReceive().AddAsync(Arg.Any<OfficeEntity>(), default);
+        }
+
+        [Fact]
+        public async Task UpdateAsync_ValidModel_ReturnUpdatedOffice()
+        {
+            //Arrange
+            var office = TestOfficeModels.Office;
+            var changedOffice = _mapper.Map<Office>(office);
+            changedOffice.Address = "New address";
+            var resultOffice = _mapper.Map<OfficeEntity>(changedOffice);
+
+            _officeRepository.UpdateAsync(Arg.Any<OfficeEntity>(), default).Returns(resultOffice);
+
+            //Act
+            var result = await _officeService.UpdateAsync(changedOffice, default);
+
+            //Assert
+            result.Address.ShouldBeEquivalentTo(changedOffice.Address);
+        }
+
+        [Fact]
+        public async Task UpdateModel_InvalidModel_ShouldNotResiveRequest()
```

Comment:
UpdateModel_InvalidModel_ShouldNotResiveRequest - why UpdateModel if testing UpdateAsync method

Reply:
i realy don't know, maybe cause of morning mood i missed it